mockgcp/mockprivilegedaccessmanager/names.go

```diff
@@ -15,43 +15,43 @@
 package mockprivilegedaccessmanager
 
 import (
+	"fmt"
 	"strings"
 
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-
-	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
 )
 
 type entitlementName struct {
-	Project       *projects.ProjectData
+	Container     string
 	Location      string
 	EntitlementID string
 }
 
 func (n *entitlementName) String() string {
-	return "projects/" + n.Project.ID + "/locations/" + n.Location + "/entitlements/" + n.EntitlementID
+	return n.parent() + "/entitlements/" + n.EntitlementID
+}
+
+func (n *entitlementName) parent() string {
+	return n.Container + "/locations/" + n.Location
 }
 
 // parseEntitlementName parses a string into a entitlementName.
-// The expected form is projects/<projectID>/locations/<region>/entitlements/<entitlementID>
+// The expected form is projects/<projectID>/locations/<region>/entitlements/<entitlementID>,
+// or folders/<folderID>/locations/<region>/entitlements/<entitlementID>, or
+// organizations/<organizationID>/locations/<region>/entitlements/<entitlementID>.
 func (s *MockService) parseEntitlementName(name string) (*entitlementName, error) {
 	tokens := strings.Split(name, "/")
 
-	if len(tokens) == 6 && tokens[0] == "projects" && tokens[2] == "locations" && tokens[4] == "entitlements" {
-		project, err := s.Projects.GetProjectByID(tokens[1])
-		if err != nil {
-			return nil, err
-		}
-
-		name := &entitlementName{
-			Project:       project,
-			Location:      tokens[3],
-			EntitlementID: tokens[5],
-		}
-
-		return name, nil
-	} else {
+	if len(tokens) != 6 ||
```

Comment:
Nit: it's often easier to express this in the affirmative, e.g.

if len(tokens) == 6 && (tokens[0] == projects || tokens[0] == folders ... {
  return OK
}
return error

Reply:
Done.